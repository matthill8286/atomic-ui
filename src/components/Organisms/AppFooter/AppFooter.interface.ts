import { FlexItemProps } from '@/components/Helper/FlexBox'
import { GroupedActionLinksProps } from '@/components/Molecules/ActionLink'
import React from 'react'

export interface AppFooterProps extends FlexItemProps {
  links?: GroupedActionLinksProps['links']
  logoUrl: string
  altText?: string
  twitterLink?: string
  instagramLink?: string
  facebookLink?: string
  youtubeLink?: string
  text?: string
  padding?: boolean
  socialIcons?: JSX.Element
}
