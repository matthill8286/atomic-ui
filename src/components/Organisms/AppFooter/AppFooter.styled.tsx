import { Cell } from '@/components/Helper/Grid/Cell'
import { FlexBox } from '@/components/Helper/FlexBox'
import { media } from '@/styles/media'
import { css, styled } from '@/styles/styled'
import { getBoxDimension } from '@/styles/sc-shared-functions'
import { BoxDimensions } from '@/types'

export const StyledPicture = styled(Cell)`
  ${media.xs} {
    width: 100%;
  }

  ${media.lg} {
    justify-self: flex-start;
    width: 100%;
  }
`

export const StyledCopyText = styled(Cell)`
  ${media.xs} {
    justify-self: flex-start;
    padding-left: ${({ theme }) => theme.spacing.base.sm};
  }

  ${media.lg} {
    justify-self: center;
  }

  ${media.xl} {
    justify-self: flex-end;
  }
`

export const StyledGroupedActionLinksWrapper = styled(Cell)`
  ${media.xs} {
    justify-self: flex-start;
  }

  ${media.lg} {
    justify-self: center;
  }
`

export const StyledFooterWrapper = styled.div<{ padding?: BoxDimensions }>(
  ({ theme, color, padding }) => css`
    ${padding ? `padding: ${getBoxDimension(theme, padding) || 0};` : ''}
  `
)

export const StyledGroupedActionLinks = styled(FlexBox)`
  ${media.xs} {
    justify-content: flex-start;
    flex-direction: column;
  }

  ${media.lg} {
    flex-direction: row;
    justify-content: space-between;
  }
`
