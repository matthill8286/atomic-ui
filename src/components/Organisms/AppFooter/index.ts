export * from './AppFooter'
export * from './AppFooter.interface'
