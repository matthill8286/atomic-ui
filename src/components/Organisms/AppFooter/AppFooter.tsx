import React from 'react'
import { Divider } from '@/components/Atoms/Divider'
import { Picture } from '@/components/Atoms/Picture'
import { CopyText } from '@/components/Atoms/Typography/CopyText'
import { FlexBox } from '@/components/Helper/FlexBox'
import { Row } from '@/components/Helper/Grid/Row'
import { useWindowDimensions } from '@/components/Helper/useWindowDimensions'
import { GroupedActionLinks } from '@/components/Molecules/ActionLink/GroupedActionLinks'
import { breakpoints } from '@/styles'
import { AppFooterProps } from './AppFooter.interface'
import {
  StyledCopyText,
  StyledPicture,
  StyledFooterWrapper,
  StyledGroupedActionLinksWrapper,
  StyledGroupedActionLinks,
} from './AppFooter.styled'
import { useTheme } from '@/utils/helper'

export const AppFooter: React.FC<AppFooterProps> = ({
  logoUrl,
  altText,
  links = [],
  text,
  socialIcons,
}): JSX.Element => {
  const { breakpoint: currentBreakpoint } = useWindowDimensions()
  const { footer } = useTheme()
  const isMobile = currentBreakpoint < breakpoints.md

  return (
    <>
      <Divider color="primary" />
      <StyledFooterWrapper padding={`${footer.padding}`}>
        <Row noMargin>
          <StyledPicture colsXl={3} colsLg={6} colsMd={8} colsSm={8} colsXs={4} align="middle">
            <FlexBox justifyContent="space-between">
              <Picture
                src={logoUrl}
                alt={altText}
                width="107px"
                height="64px"
                objectFit="contain"
              />
              {socialIcons}
            </FlexBox>
          </StyledPicture>
          <StyledGroupedActionLinksWrapper
            colsXl={6}
            colsLg={6}
            colsMd={8}
            colsSm={8}
            colsXs={4}
            align="middle">
            <StyledGroupedActionLinks>
              {links && <GroupedActionLinks links={links} />}
            </StyledGroupedActionLinks>
          </StyledGroupedActionLinksWrapper>
          <StyledCopyText colsXl={3} colsLg={12} colsMd={8} colsSm={8} colsXs={4} align="middle">
            <CopyText color={footer.color} tag="span" fontSize={isMobile ? 'xxs' : 'xs'}>
              {text}
            </CopyText>
          </StyledCopyText>
        </Row>
      </StyledFooterWrapper>
    </>
  )
}
