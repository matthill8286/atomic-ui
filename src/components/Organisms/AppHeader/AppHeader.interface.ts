import { QuickAction } from '@/components/Molecules/ActionLink'
import { Image } from '@/components/Organisms/ImageAndText'
import React from 'react'

export type HeaderContent = {
  id: string | number
  links?: QuickAction[]
  searchEnabled?: boolean
  searchLabel?: string
}

export interface AppHeaderProps {
  headerContent: HeaderContent
  searchQuery?: string
  renderSearchBar?: JSX.Element | null
  logo: Image
  onLogoClick?: (event?: React.MouseEvent) => void
}
