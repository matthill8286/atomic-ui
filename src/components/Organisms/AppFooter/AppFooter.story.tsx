import { storiesOf } from '@storybook/react'
import React from 'react'
import { AppFooter } from './AppFooter'
import { navigationListMockItems } from '@/components/Organisms/AppHeader/AppHeader.story'

storiesOf('Design System/Organisms/AppFooter', module).add('Default', () => {
  return (
    <AppFooter
      logoUrl="https://wac-cdn.atlassian.com/dam/jcr:e9ef90f9-c84a-4cba-af2f-9aa7d683ede3/Jira%20Integration.svg?cdnVersion=1324"
      instagramLink={'instagram.com'}
      twitterLink={'twitter.co.uk'}
      facebookLink={'facebook.com'}
      youtubeLink={'youtube.co.uk'}
      text="Digital Agility powered by Filtered"
      links={[...(navigationListMockItems || [])]}
    />
  )
})
