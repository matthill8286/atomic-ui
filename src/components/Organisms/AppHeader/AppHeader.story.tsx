import { action } from '@storybook/addon-actions'
import { storiesOf } from '@storybook/react'
import React from 'react'
import { LanguageMenuOption } from '@/components/Molecules/LanguageMenu'
import { SearchDropdown } from '@/components/Molecules/SearchDropdown'
import { AppHeaderComponent } from '@/components/Organisms/AppHeader'
import { QuickAction } from '@/components/Molecules/ActionLink'

export const navigationListMockItems: QuickAction[] = [
  {
    actionLabel: 'Profile',
    to: '/',
    links: [],
  },
  {
    actionLabel: 'Comments',
    to: '/',
    links: [],
  },
  {
    actionLabel: 'Goals',
    to: '/',
    links: [
      {
        actionLabel: 'Profile',
        to: '/',
        links: [],
      },
      {
        actionLabel: 'Comments',
        to: '/',
        links: [],
      },
      {
        actionLabel: 'Goals',
        to: '/',
        links: [],
      },
    ],
  },
  {
    actionLabel: 'Progress',
    to: '/',
    links: [],
  },
]

const defaultLanguages: LanguageMenuOption[] = [
  {
    id: 0,
    label: 'Dutch',
  },
  {
    id: 1,
    label: 'English',
  },
  {
    id: 2,
    label: 'French',
  },
]

const headerContent = {
  id: '',
  links: [...navigationListMockItems],
  searchBarEnabled: true,
  searchBarLabel: 'Search',
  filter: {
    enabled: true,
    placeholder: 'Filter',
  },
}

storiesOf('Design System/Organisms/AppHeader', module).add('Default', () => {
  return (
    <AppHeaderComponent
      logoRef={{ url: '', alt: '' }}
      button={{ href: '', actionLabel: 'Log Out' }}
      languages={defaultLanguages}
      renderSearchBar={
        <SearchDropdown
          index={0}
          inputValue={''}
          options={[]}
          onSelectChange={() => null}
          onInputChange={() => null}
        />
      }
      headerContent={headerContent}
      logoUrl="https://wac-cdn.atlassian.com/dam/jcr:e9ef90f9-c84a-4cba-af2f-9aa7d683ede3/Jira%20Integration.svg?cdnVersion=1324"
      onLogoClick={action('click')}
    />
  )
})
