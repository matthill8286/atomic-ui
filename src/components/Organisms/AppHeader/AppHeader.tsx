import React, { useCallback } from 'react'
import { Divider } from '@/components/Atoms/Divider'
import { Picture } from '@/components/Atoms/Picture'
import { GroupedActionLinks } from '@/components/Molecules/ActionLink'
import {
  StyledActionItems,
  StyledHeaderWrapper,
  StyledItemWrapper,
  StyledLogo,
  StyledSection,
} from '@/components/Organisms/AppHeader/AppHeader.styled'
import { AppHeaderProps } from '@/components/Organisms/AppHeader/AppHeader.interface'

export const AppHeaderComponent = React.memo(
  ({ headerContent, renderSearchBar, logo, onLogoClick }: AppHeaderProps) => {
    const internalLogoClick = useCallback(() => onLogoClick?.(), [])

    return (
      <StyledSection>
        <Divider color="primary" height="xs" />
        <StyledHeaderWrapper paddingString="sm md">
          {logo && (
            <StyledLogo>
              <Picture
                alt="thumbnail"
                onClick={() => internalLogoClick()}
                src={logo.url}
                width="107px"
                height="64px"
              />
            </StyledLogo>
          )}
          <StyledActionItems>
            {headerContent?.searchEnabled && (
              <StyledItemWrapper>{renderSearchBar}</StyledItemWrapper>
            )}
            {headerContent?.links ? (
              <StyledItemWrapper>
                <GroupedActionLinks links={[...headerContent.links]} />
              </StyledItemWrapper>
            ) : null}
          </StyledActionItems>
        </StyledHeaderWrapper>
      </StyledSection>
    )
  }
)
