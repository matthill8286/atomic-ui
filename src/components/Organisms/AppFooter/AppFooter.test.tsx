import React from 'react'
import { renderWithThemeAndRouter } from '@/testRenderer'
import { AppFooter } from './AppFooter'
import { navigationListMockItems } from '@/components/Organisms/AppHeader/AppHeader.story'

describe('Asset page footer', () => {
  it('renders with props', () => {
    const wrapper = renderWithThemeAndRouter(
      <AppFooter
        logoUrl="https://wac-cdn.atlassian.com/dam/jcr:e9ef90f9-c84a-4cba-af2f-9aa7d683ede3/Jira%20Integration.svg?cdnVersion=1324"
        text="Digital Agility powered by Filtered"
        links={[...(navigationListMockItems || [])]}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
