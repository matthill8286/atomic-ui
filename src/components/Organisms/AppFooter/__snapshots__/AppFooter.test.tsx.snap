// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Asset page footer renders with props 1`] = `
Array [
  .c0 {
  height: 8px;
  width: 100%;
  background: #00C8DC;
  border-width: 0;
  margin: 0;
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {

}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {

}

@supports (display:grid) {

}

@media (min-width:512px) {

}

@media (min-width:752px) {

}

@media (min-width:1008px) {

}

@media (min-width:1232px) {

}

@supports (display:grid) {

}

@supports (display:grid) {

}

@media (min-width:512px) {

}

@media (min-width:752px) {

}

@media (min-width:1008px) {

}

@media (min-width:1232px) {

}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {

}

@supports (display:grid) {

}

@supports (display:grid) {

}

@media (min-width:512px) {

}

@media (min-width:752px) {

}

@media (min-width:1008px) {

}

@media (min-width:1232px) {

}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {

}

@supports (display:grid) {

}

@supports (display:grid) {

}

@media (min-width:512px) {

}

@media (min-width:752px) {

}

@media (min-width:1008px) {

}

@media (min-width:1232px) {

}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {

}

@media (min-width:512px) {

}

@media (min-width:320px) {

}

@media (min-width:1008px) {

}

@media (min-width:320px) {

}

@media (min-width:1008px) {

}

@media (min-width:1232px) {

}

@media (min-width:320px) {

}

@media (min-width:1008px) {

}

@media (min-width:320px) {

}

@media (min-width:1008px) {

}

<hr
    className="c0"
    color="primary"
  />,
  .c5 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  max-width: 100%;
}

.c5 img {
  height: 64px;
  object-fit: contain;
}

.c13 {
  color: #999999;
  font-size: 14px;
  line-height: 20px;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c21 {
  color: #666666;
  font-size: 16px;
  line-height: 24px;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c25 {
  color: #666666;
  font-size: 14px;
  line-height: 20px;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  margin: 0;
  margin-bottom: 0;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  box-sizing: border-box;
  margin: calc(16px / 2);
  width: calc(100% - 16px);
  width: calc(100% - 16px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c6 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  box-sizing: border-box;
  margin: calc(16px / 2);
  width: calc(100% - 16px);
  width: calc(100% - 16px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c23 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  box-sizing: border-box;
  margin: calc(16px / 2);
  width: calc(100% - 16px);
  width: calc(100% - 16px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c19 {
  display: block;
  box-sizing: border-box;
  line-height: 1.43;
  overflow: hidden;
  box-shadow: 0 3px 4px 0 rgba(0,0,0,0.07),0 3px 3px -2px rgba(0,0,0,0.06),0 1px 8px 0 rgba(0,0,0,0.10);
  color: #000000;
  background-color: #FFFFFF;
  border: none;
  border-radius: 4px;
  height: 100%;
  width: 100%;
}

.c17 {
  position: relative;
  outline: none;
  width: 100%;
}

.c17 .Cardstyled__StyledBadgeLineWrapper-sc-9bh7mr-0 {
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
}

.c18 {
  position: absolute;
  left: -100px;
  top: 56px;
  width: 280px;
  margin: 16px 0 0 0;
  opacity: 0;
  visibility: hidden;
  z-index: 99;
  -webkit-transform: translateY(-16px);
  -ms-transform: translateY(-16px);
  transform: translateY(-16px);
  -webkit-transition: visibility 0.15s ease-in-out,opacity 0.15s ease-in-out,-webkit-transform 0.15s ease-in-out;
  -webkit-transition: visibility 0.15s ease-in-out,opacity 0.15s ease-in-out,transform 0.15s ease-in-out;
  transition: visibility 0.15s ease-in-out,opacity 0.15s ease-in-out,transform 0.15s ease-in-out;
}

.c15 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: transparent;
  border: 0;
  padding: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  color: inherit;
  opacity: 1.0;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  position: relative;
  border-bottom-color: #999999;
  padding: 16px;
  line-height: normal;
  -webkit-font-smoothing: inherit;
  -moz-osx-font-smoothing: inherit;
  -webkit-appearance: none;
}

.c15::after {
  content: '';
  position: absolute;
  display: none;
  bottom: -2px;
  height: 1px;
  background-color: #999999;
  -webkit-transform-origin: right center;
  -ms-transform-origin: right center;
  transform-origin: right center;
  -webkit-transform: scale(0,1);
  -ms-transform: scale(0,1);
  transform: scale(0,1);
  -webkit-transition: -webkit-transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
  -webkit-transition: transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
  transition: transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
}

.c15:hover::after {
  -webkit-transform-origin: left center;
  -ms-transform-origin: left center;
  transform-origin: left center;
  -webkit-transform: scale(1,1);
  -ms-transform: scale(1,1);
  transform: scale(1,1);
}

.c15::-moz-focus-inner {
  border: 0;
  padding: 0;
}

.c12 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  color: inherit;
  opacity: 1.0;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  position: relative;
  border-bottom-color: #999999;
  padding: 16px;
}

.c12::after {
  content: '';
  position: absolute;
  display: none;
  bottom: -2px;
  height: 1px;
  background-color: #999999;
  -webkit-transform-origin: right center;
  -ms-transform-origin: right center;
  transform-origin: right center;
  -webkit-transform: scale(0,1);
  -ms-transform: scale(0,1);
  transform: scale(0,1);
  -webkit-transition: -webkit-transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
  -webkit-transition: transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
  transition: transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
}

.c12:hover::after {
  -webkit-transform-origin: left center;
  -ms-transform-origin: left center;
  transform-origin: left center;
  -webkit-transform: scale(1,1);
  -ms-transform: scale(1,1);
  transform: scale(1,1);
}

.c20 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  color: inherit;
  opacity: 1.0;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  position: relative;
  border-bottom-color: #666666;
  padding: 16px;
}

.c20::after {
  content: '';
  position: absolute;
  display: none;
  bottom: -2px;
  height: 1px;
  background-color: #666666;
  -webkit-transform-origin: right center;
  -ms-transform-origin: right center;
  transform-origin: right center;
  -webkit-transform: scale(0,1);
  -ms-transform: scale(0,1);
  transform: scale(0,1);
  -webkit-transition: -webkit-transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
  -webkit-transition: transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
  transition: transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
}

.c20:hover::after {
  -webkit-transform-origin: left center;
  -ms-transform-origin: left center;
  transform-origin: left center;
  -webkit-transform: scale(1,1);
  -ms-transform: scale(1,1);
  transform: scale(1,1);
}

.c14 {
  display: inline;
  color: #999999;
}

.c22 {
  display: inline;
  color: #666666;
}

.c10 {
  list-style: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-top: 0;
  margin-right: 16px;
}

.c10:hover {
  cursor: pointer;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c11 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c11 > a {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c0 {
  padding: 8px 56px;
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {
  .c5 {
    max-width: 107px;
  }
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {
  .c5 img {
    max-width: 100%;
    width: auto;
    height: auto;
  }
}

@supports (display:grid) {
  .c1 {
    display: grid;
    grid-gap: 16px;
    grid-template-columns: repeat(4,minmax(0,1fr));
  }
}

@media (min-width:512px) {

}

@media (min-width:752px) {

}

@media (min-width:1008px) {

}

@media (min-width:1232px) {

}

@supports (display:grid) {
  .c2 {
    margin: 0;
    width: auto;
    grid-column-end: span 4;
  }
}

@supports (display:grid) {
  .c2 {
    width: auto;
    grid-column-end: span 4;
  }
}

@media (min-width:512px) {
  .c2 {
    display: block;
    margin: calc(16px / 2);
    width: calc(50% - 16px);
    width: calc(100% - 16px);
  }
}

@media (min-width:752px) {
  .c2 {
    display: block;
    margin: calc(24px / 2);
    width: calc(50% - 24px);
    width: calc(100% - 24px);
  }
}

@media (min-width:1008px) {
  .c2 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
    width: calc(50% - 24px);
  }
}

@media (min-width:1232px) {
  .c2 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
    width: calc(25% - 24px);
  }
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {

}

@supports (display:grid) {
  .c6 {
    margin: 0;
    width: auto;
    grid-column-end: span 4;
  }
}

@supports (display:grid) {
  .c6 {
    width: auto;
    grid-column-end: span 4;
  }
}

@media (min-width:512px) {
  .c6 {
    display: block;
    margin: calc(16px / 2);
    width: calc(50% - 16px);
    width: calc(100% - 16px);
  }
}

@media (min-width:752px) {
  .c6 {
    display: block;
    margin: calc(24px / 2);
    width: calc(50% - 24px);
    width: calc(100% - 24px);
  }
}

@media (min-width:1008px) {
  .c6 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
    width: calc(50% - 24px);
  }
}

@media (min-width:1232px) {
  .c6 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
    width: calc(50% - 24px);
  }
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {

}

@supports (display:grid) {
  .c23 {
    margin: 0;
    width: auto;
    grid-column-end: span 4;
  }
}

@supports (display:grid) {
  .c23 {
    width: auto;
    grid-column-end: span 4;
  }
}

@media (min-width:512px) {
  .c23 {
    display: block;
    margin: calc(16px / 2);
    width: calc(50% - 16px);
    width: calc(100% - 16px);
  }
}

@media (min-width:752px) {
  .c23 {
    display: block;
    margin: calc(24px / 2);
    width: calc(50% - 24px);
    width: calc(100% - 24px);
  }
}

@media (min-width:1008px) {
  .c23 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
    width: calc(100% - 24px);
  }
}

@media (min-width:1232px) {
  .c23 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
    width: calc(25% - 24px);
  }
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {

}

@media (min-width:512px) {
  .c17:hover .Cardstyled__StyledBadgeLineWrapper-sc-9bh7mr-0,
  .c17:hover .Cardstyled__StyledFeaturedAssetsContainer-sc-9bh7mr-3 {
    -webkit-transform: none;
    -ms-transform: none;
    transform: none;
  }
}

@media (min-width:320px) {
  .c3 {
    width: 100%;
  }
}

@media (min-width:1008px) {
  .c3 {
    justify-self: flex-start;
    width: 100%;
  }
}

@media (min-width:320px) {
  .c24 {
    justify-self: flex-start;
    padding-left: 16px;
  }
}

@media (min-width:1008px) {
  .c24 {
    justify-self: center;
  }
}

@media (min-width:1232px) {
  .c24 {
    justify-self: flex-end;
  }
}

@media (min-width:320px) {
  .c7 {
    justify-self: flex-start;
  }
}

@media (min-width:1008px) {
  .c7 {
    justify-self: center;
  }
}

@media (min-width:320px) {
  .c9 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (min-width:1008px) {
  .c9 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

<div
    className="c0"
  >
    <div
      className="c1"
    >
      <div
        className="c2 c3"
      >
        <div
          className="c4"
          display="flex"
        >
          <div
            className="c5"
            height="64px"
            width="107px"
          >
            <picture
              style={
                Object {
                  "height": "64px",
                  "width": "100%",
                }
              }
            >
              <img
                alt="image"
                height="64px"
                loading="eager"
                src="https://wac-cdn.atlassian.com/dam/jcr:e9ef90f9-c84a-4cba-af2f-9aa7d683ede3/Jira%20Integration.svg?cdnVersion=1324"
                width="107px"
              />
            </picture>
          </div>
        </div>
      </div>
      <div
        className="c6 c7"
      >
        <div
          className="c8 c9"
          display="flex"
        >
          <div
            className="c10"
          >
            <div
              className="c11"
              data-test="grouped-action-links-undefined"
            >
              <a
                className="c12"
                href="/"
                onClick={[Function]}
              >
                <span
                  className="c13 c14"
                  color="grey4"
                  fontSize="xs"
                >
                  Profile
                </span>
              </a>
            </div>
          </div>
          <div
            className="c10"
          >
            <div
              className="c11"
              data-test="grouped-action-links-undefined"
            >
              <a
                className="c12"
                href="/"
                onClick={[Function]}
              >
                <span
                  className="c13 c14"
                  color="grey4"
                  fontSize="xs"
                >
                  Comments
                </span>
              </a>
            </div>
          </div>
          <div
            className="c10"
          >
            <div
              className="c11"
              data-test="grouped-action-links-undefined"
            >
              <button
                className="c15"
                color="grey4"
                onClick={[Function]}
              >
                <span
                  className="c13 c14"
                  color="grey4"
                  fontSize="xs"
                >
                  Goals
                </span>
              </button>
              <div
                className="c16"
              >
                <div
                  className="c17 c18"
                >
                  <div
                    className="c19"
                    display="block"
                    elevation={2}
                    shape="rounded-small"
                  >
                    <div
                      className="c10"
                    >
                      <div
                        className="c11"
                        data-test="grouped-action-links-undefined"
                      >
                        <a
                          className="c20"
                          href="/"
                          onClick={[Function]}
                        >
                          <span
                            className="c21 c22"
                            color="grey5"
                            fontSize="sm"
                          >
                            Profile
                          </span>
                        </a>
                      </div>
                    </div>
                    <div
                      className="c10"
                    >
                      <div
                        className="c11"
                        data-test="grouped-action-links-undefined"
                      >
                        <a
                          className="c20"
                          href="/"
                          onClick={[Function]}
                        >
                          <span
                            className="c21 c22"
                            color="grey5"
                            fontSize="sm"
                          >
                            Comments
                          </span>
                        </a>
                      </div>
                    </div>
                    <div
                      className="c10"
                    >
                      <div
                        className="c11"
                        data-test="grouped-action-links-undefined"
                      >
                        <a
                          className="c20"
                          href="/"
                          onClick={[Function]}
                        >
                          <span
                            className="c21 c22"
                            color="grey5"
                            fontSize="sm"
                          >
                            Goals
                          </span>
                        </a>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="c10"
          >
            <div
              className="c11"
              data-test="grouped-action-links-undefined"
            >
              <a
                className="c12"
                href="/"
                onClick={[Function]}
              >
                <span
                  className="c13 c14"
                  color="grey4"
                  fontSize="xs"
                >
                  Progress
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="c23 c24"
      >
        <span
          className="c25"
          color="grey5"
          fontSize="xs"
        >
          Digital Agility powered by Filtered
        </span>
      </div>
    </div>
  </div>,
]
`;
